cpp/oneapi/dal/array.hpp

```diff
@@ -697,6 +695,15 @@ class array {
         count_ = 0;
     }
 
+    void serialize(detail::output_archive& ar) const {
+        impl_->serialize(ar);
+    }
+
+    void deserialize(detail::input_archive& ar) {
+        impl_->deserialize(ar);
+        update_data(impl_.get());
+    }
```

Comment:
A common question about serialize/deserialize: Imagine one of these operations end up with exception (for example, bad_alloc). Do we have any commitments about the object and archive state after such operation resulted with exception?

Reply:
If an exception is raised in the middle of deserialization, there is guaranties regarding valid deserialized object state, such object should not be used.